src/user/user.service.ts

```diff
@@ -4,30 +4,59 @@ import {LoginUserDto} from './dto/loginUser.dto';
 import {UpdateUserDto} from './dto/updateUser.dto';
 import {SearchUsersDto} from './dto/searchUsers.dto';
 import {UserRepository} from './userRepository';
+import {User} from 'src/entities/user';
 
 @Injectable()
 export class UserService {
     constructor(private userRepository: UserRepository) {}
 
-    async getUser(id: string): Promise<string> {
+    async getUser(id: string): Promise<User | null> {
         return this.userRepository.findById(id);
     }
 
-    async search(searchUserDto: SearchUsersDto): Promise<string[]> {
+    async search(searchUserDto: SearchUsersDto): Promise<User[] | []> {
         return this.userRepository.findAll(searchUserDto);
     }
 
-    async registerUser(registerUserDto: RegisterUserDto): Promise<string[]> {
-        return this.userRepository.findAll({login: registerUserDto.login});
-        // return { token: 'token' };
+    async registerUser(registerUserDto: RegisterUserDto): Promise<string> {
+        const user = await this.userRepository.findOne({login: registerUserDto.login});
+        if (user) {
+            //error
+            return 'user already exists.';
+        }
+        //probably phone validation is needed so there is unique phone number for each user
+
+        const userEntity = User.create(registerUserDto);
+        this.userRepository.save(userEntity);
+        return 'token';
     }
 
-    async login(loginUserDto: LoginUserDto): Promise<string[]> {
-        return this.userRepository.findAll({login: loginUserDto.login});
-        // return { token: 'token' }
+    async login(loginUserDto: LoginUserDto): Promise<string> {
+        const user = await this.userRepository.findOne({login: loginUserDto.login});
+        if (!user) {
+            //error
+            return 'user with this login was snot found.';
+        }
+
+        if (!user.verifyPassword(loginUserDto.password)) {
```

Comment:
This is something that can be implemented right now. Crypto module, etc
Also, verifying password is not User's task, use separated service instead.